### PR TITLE
Pull req - Implemented orientation support feature [rim:orientation] + Fixed Framework issue #88

### PR DESCRIPTION
**Pull request by jamesjhedley**
Implemented feature - orientation support [https://github.com/blackberry/BB10-Webworks-Packager/issues/108]
Fixed Framework issue #88 [https://github.com/blackberry/BB10-WebWorks-Framework/issues/88]
Unit tests also created.

### CODE REVIEW

**Comment by nukulb** [comment]
Please test this on both 10.0.05 and 10.0.04

**Comment by nukulb** [comment]
Please also test if you use dialog and connection APIs together does this continue to work.

**Comment by nukulb** [comment]
@jkeshavarzi - Where is the design for this? How will a user specify this in the config.xml?
Whenever you change the config.xml its important to get it reviewed by stakeholders

Also you need open an issue with the Microsite repo.
Also the issue is not properly referenced.

If the autoorientation is not working at runtime, try the browser and see that works for you

**Comment by jamesjhedley** [comment]
@nukulb Please see the change i made to config-parser.js. It works the same way it did in BlackBerry Tablet SDK as defined by product in Github feature #108 [https://github.com/blackberry/BB10-Webworks-Packager/issues/108]

Here are the docs for this feature -> https://bdsc.webapps.blackberry.com/html5/documentation/ww_developing/RIM_orientation_element_1594186_11.html

I will be opening an issue on the Microsite shortly.

File: lib/config-parser.js [hunk 1]
```diff
@@ -204,8 +204,8 @@ function processContentData(data, widgetConfig) {
 }
 
 function processOrientationData(data, widgetConfig) {
-    if (data["rim:orientation"]) {
-        var mode = data["rim:orientation"].mode;
+    if (data["rim:orientation"] && data["rim:orientation"]["@"]) {
+        var mode = data["rim:orientation"]["@"].mode;
 
         if (mode === "landscape" || mode === "portrait") {
```
Comment (jeffheifetz) [inline lib/config-parser.js:210]: Do we want to spit out a warning if they have a typo?

File: test/unit/lib/config-parser.js [hunk 2]
```diff
@@ -391,6 +391,45 @@ describe("config parser", function () {
         });
     });
```
Comment (jeffheifetz) [inline test/unit/lib/config-parser.js:393]: I think we should have a test case for the negative scenario of an invalid value.

File: lib/config-parser.js (hunk 1 above)
Comment (jamesjhedley) [inline lib/config-parser.js:210]: Seems rather excessive to spit out typo warnings in my opinion :)

Comment (jeffheifetz) [inline lib/config-parser.js:210]: I guess we don't do it for other features, it just seems like an annoying issue to find as a developer that we can easily help with. @kwallis what are your feelings?

**Comment by nukulb** [comment]
retargeted , we only need to test this on 10.0.05 builds for now.

This needs to go into next-next

File: lib/config-parser.js (hunk 1 above)
Comment (kwallis) [inline lib/config-parser.js:210]: Sounds like a pandora's box to worry about typos...

Comment (nukulb) [inline lib/config-parser.js:210]: I agree with @kwallis 
we will eventually solve this problem with a config.xml editor, this is not the place for checking typos

**Comment by nukulb** [comment]
closing this - you can open one targeting next-next

**Pull request closed without merging**